ratchet_core/src/errors.rs

```diff
@@ -255,9 +255,18 @@ impl From<InvalidHeaderValue> for Error {
 }
 
 #[derive(Clone, Copy, Error, Debug, PartialEq)]
-/// The channel is already closed
+/// The channel is closed
 #[error("The channel is already closed")]
-pub struct CloseError;
+pub enum CloseError {
+    /// The channel closed nominally. This is **not** an error and instead indicates a clean closure
+    /// of the channel by either ourselves or the peer.
+    #[error("The channel closed as expected")]
+    Nominal,
```

Comment:
Should this not be restructured if it is not an error but an indication of a clean closure of the channel?

Reply:
It could go in the `Item` enumeration but I left it out of there as that denotes actual messages that are sent during the session where as this is just denoting the end of the session.

Reply:
I think renaming it to something like:

```
enum ClosedReson {
    Stopped,
    Error,
}
```

would be better for readability (feel free to choose better names).

I just find it odd for an error enum to have a variant which is not an actual error. :)